meson.build

```diff
@@ -180,14 +187,13 @@ if host_system != 'windows'
   ]
 
   foreach h : headers
-    if cc.has_header(h)
+    if cc.has_header(h, dependencies: libbsd_overlay_dep)
```

Comment:
Do we still need this even with `add_project_dependencies(libbsd_overlay_dep, language: 'c')`? If yes, I would suggest having a separate array of dependencies:
```meson
system_deps = []

if ['gnu', 'gnu/kfreebsd'].contains(host_system)
  libbsd_overlay_dep = dependency('libbsd-overlay')
  system_deps += libbsd_overlay_dep
endif

if cc.has_header(h, dependencies: system_deps)
...
endif
```

Reply:
@ueno wrote:
> Do we still need this even with `add_project_dependencies(libbsd_overlay_dep, language: 'c')`?

Yes, both ```add_project_dependencies``` and ```add_project_arguments``` seem to be ignored by ```cc.has_header()``` et al. (Not setting '-D_GNU_SOURCE' from project_arguments when testing also makes a difference but did not cause obvious breakage)

>  If yes, I would suggest having a separate array of dependencies:
> 
> ```meson
> system_deps = []
> 
> if ['gnu', 'gnu/kfreebsd'].contains(host_system)
>   libbsd_overlay_dep = dependency('libbsd-overlay')
>   system_deps += libbsd_overlay_dep
> endif
> 
> if cc.has_header(h, dependencies: system_deps)
> ...
> endif
> ```

In my proposed change I had added libbsd-overlay only to the test-loops where it was necessary but not to e.g. ```cc.has_function('pthread_create``` or ```cc.has_function('dlopen')```. - Would you like me to basically do a s/libbsd_overlay_dep/system_deps/ or to extend the change to add system_deps to **all** instances of _cc.has..._?

cu Andreas

Reply:
The point was that the current code uses an unusual dependency declaration in the fallback code path: `libbsd_overlay_dep =  dependency('', required:false)`, which wouldn't be necessary if you build an array of dependencies (extending with the `+=` operator).

Reply:
No worries. I have switched to
```meson
system_deps = []

if ['gnu', 'gnu/kfreebsd'].contains(host_system)
  libbsd_overlay_dep = dependency('libbsd-overlay')
  system_deps += libbsd_overlay_dep
  add_project_dependencies(libbsd_overlay_dep, language: 'c')
endif
```